Composition-dependent phase transition cookbook (#5269)
geodynamics/aspect: cookbooks/composition_dependent_phase_transition/doc/composition_dependent_phase_transition.md

# Composition-dependent phase transition

*This section was contributed by Haoyuan Li, Magali Billen, Rene 
Gassmoeller and Rebecca Fildes*

The Mantle transition zone (MTZ) spans 400 - 700 km in the earth's lower upper mantle.
It contains multiple transitions, each applying hindering or assisting effects on convecting materials.

> The Composition-dependent phase transitions (CDPT) describe the phase transitions (PT) of Earth's characteristic mantle compositions.

Reviewer: ```suggestion
Composition-dependent phase transitions (CDPT) describe the phase transitions (PT) of Earth's characteristic mantle compositions.
```

> It has the virtue of simplifying numerous phases in the earth's mantle with the main dynamic effects captured (e.g. the hindering effects of the 660-km PT on the ascending and descending mantle materials).

Reviewer: ```suggestion
It has the advantage of simplifying numerous phases in the earth's mantle, while capturing the main dynamic effects such as the hindering effects of the 660-km PT on the ascending and descending mantle materials.
```

> In ASPECT, the CDPT model could be parameterized by the phase-transition interfaces in the material models.

Reviewer: ```suggestion
In ASPECT, the CDPT model is be parameterized by the phase-transition interfaces in select material models.
```

> In this cookbook, we will show an example of CDPT with the setup in {cite:t}`arredondo:2016`.

Reviewer: ```suggestion
In this cookbook, we will show an example of the CDPT implementation in ASPECT following the model design of {cite:t}`arredondo:2016`.
```

## Setup of {cite:t}`arredondo:2016`

> We include several dynamically critical phase transitions of olivine and garnet:

Reviewer: ```suggestion
The model design includes several dynamically critical phase transitions of olivine and garnet:
```

|ol -> wd|410 | 13 | 4.0 | 5.2 |
|wd -> rw|520 | 25 | 4.1 | 2.5 |
|rw -> pv + mw|660 | 5 | -2.0 | 8.4|
```

```{table} Arredondo et al., 2016: Mantle garnet phase transitions
:name: tab:garnet_phases

|Phase Transition |Depth (km)|Depth Range (km)|Clapeyron Slope (MPa / K)|Density Change (%)|Temperature Range|
| --------- | --------- | --------- | --------- | --------- | --------- |
|grt -> CaPv + grt | 560 | 60 | 4.0 | 6.7 | - |
|grt -> il| 660 | 5 | 4.0 | 5.6 | $T < 1662 K$ |
|il -> pv| 660 | 5 | -3.1 | 5.97 | $T < 1662 K$ |
|grt -> pv | 660 | 5 | 1.3 | 11.7| $T > 1662 K$ |
```

The three compositions are then compiled as a mixture of olivine and garnet.
The details of this approach are discussed in {cite:t}`arredondo:2016`

## The input file

First, we illustrate the setup with pyrolite composition.
The related entries in the prm file are listed below.
A visualization of the result is generated following the setup in Section&nbsp;{ref}`sec:cookbooks:visualizing_phase_diagram` (  {numref}`fig:pyrolite_density`).

```{literalinclude} pyrolite_material_model.prm
```

Each of the phase transitions is parameterized by their respective depth, temperature, clapeyron slope, and width.

> Moreover, two other parameters, namely the "Phase transition temperature lower limits" and the "Phase transition temperature upper limits" limit a PT to a range of temperatures, which is necessary for garnet phase transitions in the MTZ  ({numref}`fig:pyrolite_density`, the 1st, 3rd red lines and the 2nd orange line).

Reviewer: ```suggestion
Two additional parameters titled "Phase transition temperature lower limits" and the "Phase transition temperature upper limits" limit a PT to a range of temperatures, which is necessary for garnet phase transitions in the MTZ  ({numref}`fig:pyrolite_density`, the 1st, 3rd red lines and the 2nd orange line).
```

> Besides, we'll also need to make sure that the phase transitions don't intersect with each other, thus we'll need to separate the rw -> pv + mw into two parts (the 2nd red line and the 1st orange line).

Reviewer: ```suggestion
In order to ensure that the phase transitions don't intersect across P-t space, the rw -> pv + mw phase transition is separated into two parts (the 2nd red line and the 1st orange line).
```

> We can see this by dividing the deeper PTs into a cooler group (red lines) and a warmer group (orange lines).

Reviewer: ```suggestion
This can be seen by dividing the deeper PTs into a cooler group (red lines) and a warmer group (orange lines).
```

Last, in each of the temperature groups, we need to prescribe the PTs in a top-down manner: from shallower to deeper depth.

> For the careful measurement, one needs to take during this process, it's recommended to generate one phase diagram for every composition following the instructions in {ref}`sec:cookbooks:visualizing_phase_diagram` before running any dynamic models.

Reviewer: ```suggestion
In order to ensure the correct and non-intersecting phase transitions are implemented within a given model, we recommended generating gone phase diagram for every composition following the instructions in {ref}`sec:cookbooks:visualizing_phase_diagram` before running any dynamic models.
```

> You can find the whole parameter file for this phase diagram in

Reviewer: ```suggestion
The parameter file for this phase diagram is located in
```

[cookbooks/layered_phase_transition/pyrolite_phase_diagram.prm](https://www.github.com/geodynamics/aspect/blob/main/cookbooks/layered_phase_transition/pyrolite_phase_diagram.prm)

```{figure-md} fig:pyrolite_density
<img src="pyrolite_density_marked.png" style="width:90.0%" />

An example of the pyrolite composition. Red and orange marks on phase transitions demonstrate the usage of the "temperature limits": Red - low-T PTs, Orange - high-T PTs.
```

## Results

A complex setup is illustrated with a 60-degree dipping slab penetrating into the lower mantle.
In this example, an oceanic crust locates in the upper 7.5 km of the slab overlying a 37.5 km thick harzburgite layer, with the pyrolite composition assigned to the rest of the mantle ({numref}`fig:layered_initial`).
The resulting density structure is shown in figure ({numref}`fig:layered_result` a).

> In addition to densities, a set of rheology parameters could also be assigned with the same interfaces.

Reviewer: Should the above "are assigned" instead of "could also be assigned"?

Author: yes, changed
